### PR TITLE
Feature/reader discover

### PR DESCRIPTION
**Pull request by nbradbury**
This PR adds support for the upcoming "Discover" launch by our mighty Editorial team. In our mobile apps, Discover posts take two forms:

**Site Pick** - these posts highlight another blog and include a "Vist [BlogName]" attribution in both the list and detail views. Tapping this attribution opens the blog preview (site detail) for the highlighted site.

site-pick

**Editors Pick** - highlight a specific post and include a "Originally posted by [AuthorName] on [BlogName]" attribution in the list view. Tapping the post or attribution opens the **original** post in the detail view. If the original post is from a non-Jetpack self-hosted blog, the page is opened in a WebView activity instead.

editor-pick

Note to reviewer: contact me for details on viewing Discover posts. Related iOS PR [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/4018).

### CODE REVIEW

**Comment by aerych** [comment]
@nbradbury, No square icon for the blavatar in the site pick?

**Comment by nbradbury** [comment]
> No square icon for the blavatar in the site pick?

Thanks for that, fixed in 924ff7a.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 1]
```diff
@@ -119,7 +119,7 @@
 
     private static class HistoryStack extends Stack<String> {
         private final String keyName;
-        HistoryStack(String keyName) {
+        HistoryStack(@SuppressWarnings("SameParameterValue") String keyName) {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Well this is a new one to me, cool annotation.

**Comment by tonyr59h** [comment]
The Activity that shows an Editors Pick has the ActionBar title set to "WordPress" while a Site Pick sets the title to the Tag you were viewing in the Reader (i.e. "Blogs I Follow"). Seems like it should be something more specific than "WordPress", maybe the post title or the Tag being viewed.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java [hunk 2]
```diff
@@ -274,19 +299,77 @@ public void onClick(View view) {
...
     }
 
+    private void showDiscoverData(final ReaderPostViewHolder postHolder,
+                                  final ReaderPost post) {
+        final ReaderPostDiscoverData discoverData = post.getDiscoverData();
+        if (discoverData == null) {
+            postHolder.layoutDiscover.setVisibility(View.GONE);
+        }
+
+        postHolder.layoutDiscover.setVisibility(View.VISIBLE);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: Looks like the condition above was supposed to return after setting visibility.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java [hunk 3]
```diff
@@ -195,10 +196,17 @@ public void onErrorResponse(VolleyError volleyError) {
             }
         };
 
-        WordPress.getRestClientUtilsV1_1().get(sb.toString(), null, null, listener, errorListener);
+        // TODO: as of 09-July-2015 there is no v1.2 endpoint for Freshly Pressed, so we have to
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java]: We should probably file an issue to remind us to use the v1.2 endpoint exclusively when it supports FP.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import android.text.TextUtils;
 
 import com.android.volley.VolleyError;
+import com.wordpress.rest.RestClient;
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java]: nitpick: unused import

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java [hunk 5]
```diff
@@ -308,10 +329,10 @@ private String formatPostContentForWebView(final String content) {
         .append("  p, div, li { line-height: 1.6em; font-size: 0.95em; }")
         .append("  h1, h2 { line-height: 1.2em; }")
 
-        // counteract pre-defined height/width styles
+                // counteract pre-defined height/width styles
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java]: nitpick: weird indenting

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java (hunk 2 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: > Looks like the condition above was supposed to return after setting visibility.

Good catch! Fixed in 5541b514d2.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java (hunk 3 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java]: > We should probably file an issue to remind us to use the v1.2 endpoint exclusively when it supports FP.

Filed as #2991.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java (hunk 4 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java]: > nitpick: unused import

Corrected in 18b2939caf.

**Comment by nbradbury** [comment]
> The Activity that shows an Editors Pick has the ActionBar title set to "WordPress" while a Site Pick sets the title to the Tag you were viewing in the Reader

Fixed in db6fa7f.

**Comment by nbradbury** [comment]
@tonyr59h Thanks for catching some problems in the initial PR! Ready for round two now.

**Comment by tonyr59h** [comment]
Nice job! No more taking PR's before vacation for me :sweat_smile:

:shipit:

**Pull request merged**